src/get-started/first-java-service.md

```diff
@@ -0,0 +1,618 @@
+the transactions to our service:
+
+```shell
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  keygen
+```
+
+Now, try to submit transactions adding your own vehicles to the blockchain:
+
+```shell
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  add-vehicle -a -n=test-car-registry "My car" "VW" "Polo" "$USER"
+```
+
+`-a` option requests the client to _await_ the transaction commitment, so that
+we can see its execution result; `-n` specifies the service instance _name_
+that we assigned on its start.
+
+Check they are in the registry:
+
+```shell
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  find-vehicle -n=test-car-registry "My car"
+```
+
+Suppose for a minute, that Emmett Brown has got tired of time travels,
+and decided to transfer his DeLorean to you:
+
+```shell
+# Check the entry beforehand
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  find-vehicle -n=test-car-registry "V2"
+
+# Change the owner to the current user
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  change-owner -a -n=test-car-registry "V2" "$USER"
+
+# See the updated entry
+java -jar car-registry-client/target/car-registry-client-1.0.0-SNAPSHOT.jar \
+  find-vehicle -n=test-car-registry "V2"
+```
+
+!!! success
+    Congratulations! You have successfully implemented a simple Exonum
+    service, started a network of nodes, and deployed your application
+    in it!
+
+<!--
+TODO: Are there any articles that go well after this tutorial completion
+that we shall mention at its end?
+-->
+
+## Exercises
+
+**E1**. The transaction [transferring the ownership over a vehicle](#transfer-the-ownership-transaction)
+currently allows transferring it to the same owner
+(e.g., "John Doe" > "John Doe"). It also accepts empty owner field.
+Modify its code to forbid such input arguments.
+
+**E2**. Try the following sequence of operations with a fresh service state:
+
+1. Change the owner of vehicle "V1" to "John Doe"
+2. Change the owner of vehicle "V1" to yourself
+3. Change the owner of vehicle "V1" back to "John Doe".
+
+The third operation is expected to be rejected at submission because
+the corresponding transaction message is equal to the first
+transaction message, which is already committed. Exonum rejects transactions
+with the same messages (basically, with same arguments and coming from
+the same author) to prevent their replication by another user.
+Modify the transaction so that such operation is possible.
+
+??? help "Hint"
+    A common approach to make transactions with the same arguments from the
+    same author have different messages is to include a _seed_ field. Each
+    transaction author will have to set it to a unique value (to that author
+    and set of arguments). As a seed, each author may use a counter of submitted
+    transactions, or a random value.
```

Comment:
nit: date (or unixtime) also seems like a pretty obvious solution

Reply:
> or the current time in ms if that author does not create transactions in rapid succession/several transactions in ms.

That's a good idea, but it required this clause that I think we shall probably skip that option.

Reply:
Probably, yes.